dataset/src/test/scala/frameless/functions/AggregateFunctionsTests.scala

```diff
@@ -357,4 +358,52 @@ class AggregateFunctionsTests extends TypedDatasetSuite {
     check(forAll(prop[Vector[Long]] _))
     check(forAll(prop[BigDecimal] _))
   }
+
+  test("corr") {
+    val spark = session
+    import spark.implicits._
+    import frameless.CatalystCast._
+
+    def prop[A: TypedEncoder, B: TypedEncoder](xs: List[X3[Int, A, B]])(
+      implicit
+      encEv:Encoder[(Int, A, B)],
+      evCanBeDoubleA:CatalystCast[A, Double],
+      evCanBeDoubleB:CatalystCast[B, Double]
+    ): Prop = {
+
+      // mapping with this function is needed because spark uses Double.NaN for some semantics in the correlation function. ?= for prop testing will use == underlying and will break because Double.NaN != Double.NaN
+      val nanHandler : Double => Option[Double] = value => if (!value.equals(Double.NaN)) Option(value) else None
+      //making sure that null => None and does not result in 0.0d because of row.getAs[Double]'s use of .asInstanceOf
+      val nanNullHandler : Any => Option[Double] = value=> if (value != null) nanHandler(value.asInstanceOf[Double]) else None
+
+      val tds = TypedDataset.create(xs)
+      val tdCorrelation = tds.groupBy(tds('a)).agg(corr(tds('b), tds('c)))
+        .map(
+          kv => (kv._1, kv._2.flatMap(nanNullHandler))
```

Comment:
I'm a bit confused by this test, it weird to have to apply this `nanNullHandler` to both the typed and the untyped dataset. I would expect it either only on the typed version to flatten the `Option[Double]`, or only on the untyped version to lift the `Double` into a `Option[Double]`